llama_index/llms/ollama.py

```diff
@@ -0,0 +1,150 @@
+from typing import Any, Callable, Dict, Optional, Sequence, List, Iterator
+import json
+from llama_index.bridge.pydantic import Field, PrivateAttr
+
+from llama_index.callbacks import CallbackManager
+from llama_index.constants import DEFAULT_CONTEXT_WINDOW, DEFAULT_NUM_OUTPUTS
+from llama_index.llms.base import (
+    ChatMessage,
+    ChatResponse,
+    ChatResponseGen,
+    CompletionResponse,
+    CompletionResponseGen,
+    LLMMetadata,
+    llm_chat_callback,
+    llm_completion_callback,
+)
+from llama_index.llms.custom import CustomLLM
+from llama_index.llms.generic_utils import completion_response_to_chat_response
+from llama_index.llms.generic_utils import (
+    messages_to_prompt as generic_messages_to_prompt,
+)
+from llama_index.llms.generic_utils import stream_completion_response_to_chat_response
+
+
+class Ollama(CustomLLM):
+    base_url: str = "http://localhost:11434"
```

Comment:
Should this be a configurable parameter in the init?

Reply:
all of ollama's model endpoint is exposed at this port: https://ollama.ai/library/codellama
It's actually hardcoded in their code: https://github.com/jmorganca/ollama/blob/ccc3e9ac6db7e7e44bc747916792188e17251cdf/api/client.go#L71

Reply:
Interesting choice for them haha but makes sense!